libibumad/umad.c

```diff
@@ -618,9 +627,13 @@ int umad_open_port(const char *ca_name, int portnum)
 {
 	char dev_file[UMAD_DEV_FILE_SZ];
 	int umad_id, fd;
+	unsigned int abi_version = get_abi_version();
 
 	TRACE("ca %s port %d", ca_name, portnum);
 
+	if (!abi_version)
+		return -EOPNOTSUPP;
```

Comment:
This patch looks right to me but I want to test this with the various tools to be sure and won't be able to get to that until either Monday or Tuesday.

Reply:
I ran some tests using OpenSM and infiniband-diags and this is working as expected.

Only thing it does is change when error occurs. It used to occur at umad_init time which is always the first thing and now won't occur until first umad_open_port. This might surprise some users but don't see a way around that.

:+1